### PR TITLE
feat: support single node model

### LINKED ISSUE #225: pod controller support single node model
**What would you like to be added**:
pod controller support single node model

**Why is this needed**:
pod controller support single node model

### PR DESCRIPTION
**Pull request by OrangeBao**
#### What type of PR is this?

/kind feature

#### Which issue(s) does this PR fix?

Fixes #225
#### Does this PR introduce a user-facing change?

```release-note
make pod support single node model
```

### CODE REVIEW

File: pkg/clustertree/cluster-manager/controllers/pod/root_pod_controller.go [hunk 1]
```diff
@@ -212,11 +212,16 @@ func (r *RootPodReconciler) SetupWithManager(mgr manager.Manager) error {
 		}
 
 		p := obj.(*corev1.Pod)
-		// skip daemonset
+		// ignore the pod who donnot has the annotations "kosmos-io/owned-by-cluster"
+		// TODO： use const
+		if _, ok := p.Annotations["kosmos-io/owned-by-cluster"]; !ok {
```
Comment (xyz2277) [inline pkg/clustertree/cluster-manager/controllers/pod/root_pod_controller.go]: this annotation item doesn't belong pod, it's in node

File: pkg/clustertree/cluster-manager/controllers/pod/root_pod_controller.go [hunk 2]
```diff
@@ -212,11 +212,16 @@ func (r *RootPodReconciler) SetupWithManager(mgr manager.Manager) error {
...
+			return false
+		}
 		if !strings.HasPrefix(p.Spec.NodeName, utils.KosmosNodePrefix) {
```
Comment (xyz2277) [inline pkg/clustertree/cluster-manager/controllers/pod/root_pod_controller.go]: int 1vs1 mode, pod should not be skiped

File: pkg/clustertree/cluster-manager/utils/leaf_resource_manager.go [hunk 3]
```diff
@@ -16,57 +18,136 @@ var (
...
+func has(arry []ClusterNode, target string) bool {
+	for _, v := range arry {
+		if v.NodeName == target {
+			return true
+		}
+	}
+	return false
+}
+
+func getClusterNode(arry []ClusterNode, target string) *ClusterNode {
```
Comment (xyz2277) [inline pkg/clustertree/cluster-manager/utils/leaf_resource_manager.go]: arry is not a good name

File: .gitignore [hunk 4]
```diff
@@ -28,4 +28,6 @@ _delete/
 
 kube-config
 
-__debug_bin*
\ No newline at end of file
+__debug_bin*
+
+ignore_dir
```
Comment (wuyingjun-lucky) [inline .gitignore]: Add a new line

File: pkg/clustertree/cluster-manager/controllers/pod/root_pod_controller.go [hunk 5]
```diff
@@ -272,24 +282,12 @@ func (r *RootPodReconciler) createStorageInLeafCluster(ctx context.Context, lr *
...
-				}
-				if secretObj.Type == corev1.SecretTypeServiceAccountToken {
-					if err := r.createServiceAccountInLeafCluster(ctx, lr, secretObj); err != nil {
-						klog.Error(err)
-						return err
-					}
-				}
+
+			if err := storageHandler.BeforeCreateInLeaf(ctx, r, lr, unstructuredObj, rootpod); err != nil {
+				return err
```
Comment (wuyingjun-lucky) [inline pkg/clustertree/cluster-manager/controllers/pod/root_pod_controller.go:307]: storageHandler seems better than judging the gvr

File: pkg/utils/podutils/pod.go [hunk 6]
```diff
@@ -166,7 +167,9 @@ func FitPod(pod *corev1.Pod, ignoreLabels []string) *corev1.Pod {
 	podCopy.Spec.NodeName = ""
 	podCopy.Status = corev1.PodStatus{}
 
-	podCopy.Spec.SchedulerName = ""
+	if cleanNodeName {
+		podCopy.Spec.SchedulerName = ""
+	}
```
Comment (xyz2277) [inline pkg/utils/podutils/pod.go:173]: why SchedulerName？

**Pull request merged**